lean-toolchain

```diff
@@ -1 +1 @@
-leanprover/lean4:v4.2.0-rc3
+leanprover/lean4:v4.2.0-rc4
```

Comment:
Why is this appearing in the diff? Master is on `rc4`. In any case LGTM, thanks!

bors d+